pycbc/filter/matchedfilter.py

```diff
@@ -1229,8 +1229,263 @@ def quadratic_interpolate_peak(left, middle, right):
+            if result is None: break
+            results.append(result)
+            veto_info += veto
+
+        result = self.combine_results(results)
+        
+        if self.max_triggers_in_batch:
+            sort = result['snr'].argsort()[::-1][:self.max_triggers_in_batch]
+            for key in result:
+                result[key] = result[key][sort]
+            
+            tmp = veto_info
+            veto_info = [tmp[i] for i in sort]
+        
+        result = self._process_vetoes(result, veto_info)
+        return result
+
+    def _process_vetoes(self, results, veto_info):
+        """Calculate signal based vetoes"""            
+        chisq = numpy.array(numpy.zeros(len(veto_info)), numpy.float32, ndmin=1)
+        dof = numpy.array(numpy.zeros(len(veto_info)), numpy.uint32, ndmin=1)
+        results['chisq'] = chisq
+        results['chisq_dof'] = dof
+        
+        keep = []
+        for i, (snrv, norm, l, htilde, stilde) in enumerate(veto_info): 
+            correlate(htilde, stilde, htilde.cout)
+            c, d = self.power_chisq.values(htilde.cout, snrv,
+                                           norm, stilde.psd, [l], htilde)
+            chisq[i] = c[0] / d[0]
+            dof[i] = d[0]
+            
+            if self.newsnr_threshold:
+                newsnr = events.newsnr(results['snr'][i], chisq[i])
+                if newsnr >= self.newsnr_threshold:
+                    keep.append(i)
+                    
+        if self.newsnr_threshold:
+            keep = numpy.array(keep, dtype=numpy.uint32)
+            for key in results:
+                results[key] = results[key][keep]
+                
+        return results 
+
+    def _process_batch(self):
+        """Process only a single batch group of data"""
+        if self.block_id == len(self.tgroups):
+            return None, None
+
+        tgroup = self.tgroups[self.block_id]
+        psize = self.chunk_tsamples[self.block_id]
+        mid = self.mids[self.block_id]
+        stilde = self.data.overwhitened_data(tgroup[0].delta_f)
+        psd = stilde.psd 
+
+        valid_end = int(psize - self.data.trim_padding)
+        valid_start = int(valid_end - self.data.blocksize * self.data.sample_rate)
+
+        seg = slice(valid_start, valid_end)
+        
+        self.corr[self.block_id].execute(stilde)
+        self.ifts[mid].execute()
+
+        snr = numpy.zeros(len(tgroup), dtype=numpy.complex64)
+        time = numpy.zeros(len(tgroup), dtype=numpy.float64)
+        templates = numpy.zeros(len(tgroup), dtype=numpy.uint64)
+        sigmasq = numpy.zeros(len(tgroup), dtype=numpy.float32)
+
+        time[:] = self.data.start_time
+
+        result = {}
+        tkeys = tgroup[0].params.dtype.names
+        for key in tkeys:
+            result[key] = []
+
+        veto_info = []
+
+        # Find the peaks in our SNR times series from the various templates
+        i = 0
+        for htilde in tgroup:
```

Comment:
It's not incremented each time actually. Only if something is above
threshold, otherwise, not.

On Thu, Aug 4, 2016 at 3:27 AM, Christopher M. Biwer <
notifications@github.com> wrote:

> In pycbc/filter/matchedfilter.py
> <https://github.com/ligo-cbc/pycbc/pull/1002#discussion_r73450288>:
>
> > +        time = numpy.zeros(len(tgroup), dtype=numpy.float64)
> > +        templates = numpy.zeros(len(tgroup), dtype=numpy.uint64)
> > +        sigmasq = numpy.zeros(len(tgroup), dtype=numpy.float32)
> > +
> > +        time[:] = self.data.start_time
> > +
> > +        result = {}
> > +        tkeys = tgroup[0].params.dtype.names
> > +        for key in tkeys:
> > +            result[key] = []
> > +
> > +        veto_info = []
> > +
> > +        # Find the peaks in our SNR times series from the various templates
> > +        i = 0
> > +        for htilde in tgroup:
>
> Looks like your incrementing i after each iteration. Why not do for
> i,htilde in enumerate(tgroup)?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ligo-cbc/pycbc/pull/1002/files/927835a0c99d219b5ac22a9beb17aa58ff9c4b48#r73450288>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACGrRvPoHgdArj1OST61rU_eD7I8E2Kwks5qcUAGgaJpZM4JZwrO>
> .
>



-- 
Dr. Alexander Nitz
Max Planck Institute for Gravitational Physics (Albert Einstein Institute)
Callinstrasse 38
D-30167 Hannover,  Germany
Tel: +49 511 762-17097